Profiler/log/AttachLog.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include "FileLogBase.h"
+
+/**
+ * Manages a log file on the file system to which profiler attach events (time, executable name and process ID) are written.
+ * Unless mentioned otherwise, all methods in this class are thread-safe and perform their own synchronization.
+ */
+class AttachLog: public FileLogBase

```

Comment:
`AttachLog` is not really a `FileLogBase`... you don't use any of the methods declared by the supertype on the instances for `AttachLog`. I would suggest to implement `AttachLog` a separate and use a `FileLog` internally to implement it using delegation.

Reply:
I'd rather move some of the log methods from `FileLogBase` to `TraceLog`. I think conceptionally `AttachLog` should be a `FileLogBase` and using delegation here would require a not insignificant amount of restructuring, which probably results in remove FileLogBase all together and moving it's functionality into Utils. Although possible, I'd prefer the solution mentioned at the start of the comment. What do you think?

Reply:
I'd be ok with that, I guess :)

Reply:
Moved all log methods apart from `writeToFile` and `writeTupleToFile` which are needed in both attach and trace log.